src/states/elder/mod.rs

```diff
@@ -965,29 +969,42 @@ impl Elder {
         // If the message is to a single node and we have the connection info for this node, don't
         // go through the routing table
         let single_target = if let Authority::Node(node_name) = dst {
-            self.peer_map.get_id(&node_name)
+            self.chain.get_p2p_node(&node_name)
         } else {
             None
         };
 
-        let (target_pub_ids, dg_size) = if let Some(target) = single_target {
-            (vec![*target], 1)
+        let (target_p2p_nodes, dg_size) = if let Some(target) = single_target {
+            (vec![target.clone()], 1)
         } else {
-            self.get_targets(signed_msg.routing_message())?
+            // WIP: neet to get targets without using the peer_map (get_targets uses peer_map
+            // internally)
+            let (targets, dg_size) = self.get_targets(signed_msg.routing_message())?;
+            (
```

Comment:
I'm not sure about that WIP.
It seems get_tragets use peer_map to find connected, also we are still using get_targets, so not clear what we are doing here.

Question, why would we not get rid of is_connected from within get_targets + chain::targets?
All nodes that can be targets are elders now, so it seem we find them in our or our neighbour'elder then we don't have to check for is_connected.

Additionally, you can also replace the return type from PublicId to P2pNodes. (though you could keep the return type as is in this PR, and use self.chain.get_p2p_node. so we do not use peer_map here.

Reply:
Yes that's right, we do so in part 3. The reason that commit is in part 3 and not part 2 is that I've not yet investigated the test failures I get from it

Reply:
Ok, that seem weird I did not expect a test failure here, can you point me to the commit you do it?

Reply:
https://github.com/maidsafe/routing/pull/1885/commits/bf3a29ce4453b3359e91f76dfc52edcd52f4cca6

Reply:
Thanks, I'll have a look tomorrow.

Reply:
I had a look at it and with a little tweaking I got the tests to pass

Reply:
Great :-)